Source/DataSources/I3SDataSource.js

```diff
@@ -0,0 +1,4601 @@
+          }
+        }
+      }
+    }
+
+    if (this._parent._data.mesh) {
+      texturePath =
+        this._layer._completeUriWithoutQuery +
+        "/nodes/" +
+        this._parent._data.mesh.material.resource +
+        "/textures/" +
+        textureName +
+        query;
+    }
+  }
+
+  var gltfTextures = [];
+  var gltfImages = [];
+  var gltfSamplers = [];
+
+  if (isTextured) {
+    gltfTextures = [
+      {
+        sampler: 0,
+        source: 0,
+      },
+    ];
+
+    gltfImages = [
+      {
+        uri: texturePath,
+      },
+    ];
+
+    gltfSamplers = [
+      {
+        magFilter: 9729,
+        minFilter: 9986,
+        wrapS: 10497,
+        wrapT: 10497,
+      },
+    ];
+
+    gltfMaterial.pbrMetallicRoughness.baseColorTexture.index = 0;
+  }
+
+  var gltfData = {
+    scene: 0,
+    scenes: [
+      {
+        nodes: nodesInScene,
+      },
+    ],
+    nodes: nodes,
+    meshes: meshes,
+    buffers: buffers,
+    bufferViews: bufferViews,
+    accessors: accessors,
+    materials: [gltfMaterial],
+    textures: gltfTextures,
+    images: gltfImages,
+    samplers: gltfSamplers,
+    asset: {
+      version: "2.0",
+    },
+  };
+
+  return gltfData;
+};
+
+/*
+..######..##.....##..######..########..#######..##.....##
+.##....##.##.....##.##....##....##....##.....##.###...###
+.##.......##.....##.##..........##....##.....##.####.####
+.##.......##.....##..######.....##....##.....##.##.###.##
+.##.......##.....##.......##....##....##.....##.##.....##
+.##....##.##.....##.##....##....##....##.....##.##.....##
+..######...#######...######.....##.....#######..##.....##
+
+.##.....##..#######...#######..##....##..######.
+.##.....##.##.....##.##.....##.##...##..##....##
+.##.....##.##.....##.##.....##.##..##...##......
+.#########.##.....##.##.....##.#####.....######.
+.##.....##.##.....##.##.....##.##..##.........##
+.##.....##.##.....##.##.....##.##...##..##....##
+.##.....##..#######...#######..##....##..######.
+*/
+
+// Reimplement Cesium3DTile.prototype.requestContent so that
+// We get a chance to load our own b3dm from I3S data
+Cesium3DTile.prototype._hookedRequestContent =
+  Cesium3DTile.prototype.requestContent;
+
+/**
+ * @private
+ */
+Cesium3DTile.prototype._resolveHookedObject = function () {
+  var that = this;
+  // Keep a handle on the early promises
+  var _contentReadyToProcessPromise = this._contentReadyToProcessPromise;
+  var _contentReadyPromise = this._contentReadyPromise;
+
+  // Call the real requestContent function
+  this._hookedRequestContent();
+
+  // Fulfill the promises
+  if (_contentReadyToProcessPromise) {
+    this._contentReadyToProcessPromise.then(function () {
+      _contentReadyToProcessPromise.resolve();
+    });
+  }
+
+  if (_contentReadyPromise) {
+    this._contentReadyPromise.then(function () {
+      that._isLoading = false;
+      that._content._contentReadyPromise.resolve();
+    });
+  }
+};
+
+Cesium3DTile.prototype.requestContent = function () {
```

Comment:
Instead of re-implementations, it may be useful to modify the original function to support callbacks, that classes like these can choose to work with.